### PR TITLE
Catalog code cleanup

### PR DESCRIPTION
**Pull request by tli2**
Addresses issue #1398.

This PR fixes 1, 2 and renames CatalogObjects to Entries. Other minor code style fixes are included as well.

### CODE REVIEW

File: src/binder/binder_context.cpp [hunk 1]
```diff
@@ -38,8 +38,10 @@ void BinderContext::AddRegularTable(const std::string db_name,
                                     const std::string table_alias,
                                     concurrency::TransactionContext *txn) {
   // using catalog object to retrieve meta-data
-  auto table_object = catalog::Catalog::GetInstance()->GetTableObject(
-      db_name, schema_name, table_name, txn);
+  auto table_object = catalog::Catalog::GetInstance()->GetTableObject(txn,
```
Comment (apavlo) [inline src/binder/binder_context.cpp]: We should probably rename this as `GetTableEntry` too.

File: src/catalog/abstract_catalog.cpp [hunk 2]
```diff
@@ -71,21 +71,27 @@ AbstractCatalog::AbstractCatalog(const std::string &catalog_table_ddl,
   auto catalog_database_name = create_plan->GetDatabaseName();
   PELOTON_ASSERT(catalog_schema_name == std::string(CATALOG_SCHEMA_NAME));
   // create catalog table
-  Catalog::GetInstance()->CreateTable(
-      catalog_database_name, catalog_schema_name, catalog_table_name,
-      std::unique_ptr<catalog::Schema>(catalog_table_schema), txn, true);
+  Catalog::GetInstance()->CreateTable(txn,
```
Comment (apavlo) [inline src/catalog/abstract_catalog.cpp]: Unless this function also creates the `DataTable` object (which it shouldn't), maybe we should rename this to `CreateTableEntry`.

File: src/include/catalog/database_catalog.h [hunk 3]
```diff
@@ -104,10 +110,9 @@ class DatabaseCatalog : public AbstractCatalog {
   ~DatabaseCatalog();
 
   // Global Singleton, only the first call requires passing parameters.
-  static DatabaseCatalog *GetInstance(
-      storage::Database *pg_catalog = nullptr,
-      type::AbstractPool *pool = nullptr,
-      concurrency::TransactionContext *txn = nullptr);
+  static DatabaseCatalog *GetInstance(concurrency::TransactionContext *txn = nullptr,
```
Comment (apavlo) [inline src/include/catalog/database_catalog.h]: Can we remove the default values?

File: src/catalog/catalog.cpp [hunk 4]
```diff
@@ -65,12 +65,14 @@ Catalog::Catalog() : pool_(new type::EphemeralPool()) {
...
   // Create catalog tables
-  DatabaseCatalog::GetInstance(peloton, pool_.get(), txn);
-  BootstrapSystemCatalogs(peloton, txn);
+  DatabaseCatalog::GetInstance(txn, peloton, pool_.get());
+  BootstrapSystemCatalogs(txn, peloton);
 
   // Insert peloton database into pg_database
-  DatabaseCatalog::GetInstance()->InsertDatabase(
-      CATALOG_DATABASE_OID, CATALOG_DATABASE_NAME, pool_.get(), txn);
+  DatabaseCatalog::GetInstance(nullptr, nullptr, nullptr)->InsertDatabase(txn,
```
Comment (apavlo) [inline src/catalog/catalog.cpp]: Why pass a null txn pointer to `DatabaseCatalog::GetInstance()` when you actually have the txn pointer?

File: src/catalog/catalog.cpp [hunk 5]
```diff
@@ -674,33 +848,36 @@ ResultType Catalog::DropDatabaseWithOid(oid_t database_oid,
...
+ResultType Catalog::DropSchema(concurrency::TransactionContext *txn,
+                               const std::string &database_name,
+                               const std::string &schema_name) {
   if (txn == nullptr)
     throw CatalogException("Do not have transaction to drop schema " +
                            schema_name);
 
   auto database_object =
-      DatabaseCatalog::GetInstance()->GetDatabaseObject(database_name, txn);
+      DatabaseCatalog::GetInstance(nullptr,
```
Comment (apavlo) [inline src/catalog/catalog.cpp]: Same here. You actually have the txn pointer.

File: src/catalog/catalog.cpp [hunk 6]
```diff
@@ -1099,23 +1316,29 @@ void Catalog::AddBuiltinFunction(
  * @param   code_context  the code_context that holds the generated LLVM
  *                        query code
  */
-void Catalog::AddPlpgsqlFunction(
-    const std::string &name, const std::vector<type::TypeId> &argument_types,
-    const type::TypeId return_type, oid_t prolang, const std::string &func_src,
-    std::shared_ptr<peloton::codegen::CodeContext> code_context,
-    concurrency::TransactionContext *txn) {
+void Catalog::AddPlpgsqlFunction(concurrency::TransactionContext *txn,
```
Comment (apavlo) [inline src/catalog/catalog.cpp]: I don't think that this should be called `AddPlpgsqlFunction`. You are passing in the `prolang` argument, so it should just be called `AddFunction`, right? Furthermore, we are are referring to UDFs as procedures (i.e., `pg_proc` table), so it really should be called `AddProcedure`.

File: src/catalog/catalog.cpp [hunk 7]
```diff
@@ -1188,14 +1411,16 @@ void Catalog::InitializeLanguages() {
     auto &txn_manager = concurrency::TransactionManagerFactory::GetInstance();
     auto txn = txn_manager.BeginTransaction();
     // add "internal" language
-    if (!LanguageCatalog::GetInstance().InsertLanguage("internal", pool_.get(),
-                                                       txn)) {
+    if (!LanguageCatalog::GetInstance().InsertLanguage(txn,
```
Comment (apavlo) [inline src/catalog/catalog.cpp]: This is not your problem, but we should **not** be initializing the language table in this function.

File: src/catalog/catalog.cpp (hunk 6 above)
Comment (tli2) [reply src/catalog/catalog.cpp]: I had no idea what this is supposed to do. Will change.

File: src/catalog/catalog.cpp (hunk 7 above)
Comment (tli2) [reply src/catalog/catalog.cpp]: Ack

Comment (tli2) [reply src/catalog/catalog.cpp]: Done. #1421

File: src/binder/binder_context.cpp (hunk 1 above)
Comment (apavlo) [reply src/binder/binder_context.cpp]: We should vote on this. I think `Get*CatalogEntry` would be best.

Comment (ksaito7) [reply src/binder/binder_context.cpp]: I vote `Get*CatalogEntry`. I think the accuracy of function name is more important than length. And I think this name is not so long.

File: src/include/catalog/language_catalog.h [hunk 8]
```diff
@@ -63,28 +63,29 @@ class LanguageCatalog : public AbstractCatalog {
...
+                                                         oid_t lang_oid) const;
 
-  std::unique_ptr<LanguageCatalogObject> GetLanguageByName(
-      const std::string &lang_name, concurrency::TransactionContext *txn) const;
+  std::unique_ptr<LanguageCatalogEntry> GetLanguageByName(concurrency::TransactionContext *txn,
+                                                          const std::string &lang_name) const;
 
   enum ColumnId {
     OID = 0,
     LANNAME = 1,
```
Comment (pervazea) [inline src/include/catalog/language_catalog.h]: Poor abbreviation. Should be LANG or LANGUAGE

File: src/binder/binder_context.cpp (hunk 1 above)
Comment (pervazea) [reply src/binder/binder_context.cpp]: I'd go with apavlo, ksaito7 and vote for ... Entry.

Comment (tli2) [reply src/binder/binder_context.cpp]: Fine, I'll rename them to GetxxxCatalogEntry

File: src/catalog/abstract_catalog.cpp (hunk 2 above)
Comment (tli2) [reply src/catalog/abstract_catalog.cpp]: Ironically, I think it does create the DataTable object (Not that it should)

**Pull request merged**